src/index.js

```diff
@@ -28,27 +60,38 @@ class UglifyJsPlugin {
 		compiler.plugin("compilation", (compilation) => {
 			if(options.sourceMap) {
 				compilation.plugin("build-module", (module) => {
-					// to get detailed location info about errors
+          // to get detailed location info about errors
 					module.useSourceMap = true;
 				});
 			}
 			compilation.plugin("optimize-chunk-assets", (chunks, callback) => {
-				const files = [];
-				chunks.forEach((chunk) => files.push.apply(files, chunk.files));
-				files.push.apply(files, compilation.additionalChunkAssets);
-				const filteredFiles = files.filter(ModuleFilenameHelpers.matchObject.bind(undefined, options));
-				filteredFiles.forEach((file) => {
-					const oldWarnFunction = uglify.AST_Node.warn_function;
-					const warnings = [];
-					let sourceMap;
+				const files = compilation.additionalChunkAssets;
+				let i = chunks.length - 1;
+				while(i--) {
+          // eslint-disable-next-line prefer-spread
+					files.push.apply(files, chunks[i].files);
+				}
+
+				let file;
+				let sourceMap;
+				let input;
+				let inputSourceMap;
+				let map;
+				let oldWarnFunction;
```

Comment:
Better to write a comment explaining why all these variables were hoisted outside the loop.

It's usually not a good idea to write variables outside the scope where they are needed. I assume this was done to avoid `transform-block-scoping` creating an anonymous function inside the loop for correct scoping.